pkg/tsdb/v3iotsdb_integration_test.go

```diff
@@ -158,13 +201,26 @@ func testIngestDataCase(t *testing.T, testParams tsdbtest.TestParams) {
 		}
 
 		if _, err := appender.WaitForCompletion(0); err != nil {
-			t.Fatalf("Failed to wait for appender completion. reason: %s", err)
+			if !isExpected(testParams, err) {
+				t.Fatalf("Failed to wait for appender completion. reason: %s", err)
+			}
 		}
 
-		tsdbtest.ValidateCountOfSamples(t, adapter, dp.Name, len(dp.Data), from, to, -1)
+		expectedCount := len(dp.Data)
```

Comment:
i would prefer validating the values+count.
You can potentially add an `expectedData` rather then `expectedCount` and validate it against the actual data.
if `expectedData` is not set. expected will be the same as the "ingested" data.

Reply:
ok, so do it. I don't think that this PR should contain everything you miss in this project :-)